src/hooks/core/push-notification/usePushNotificationHook.ts

```diff
@@ -129,23 +129,29 @@ const usePushNotificationHook = () => {
   };
 
   const __handleNotification = async (notification) => {
-    if (notification.data.type === 'feed' || notification.data.type === 'reaction') {
+    if (notification?.data?.type === 'topic' && notification?.data?.topic_name) {
+      navigation.navigate('TopicPageScreen', {
+        id: notification?.data?.topic_name
+      });
+    }
+
+    if (notification?.data?.type === 'feed' || notification?.data?.type === 'reaction') {
       navigation.navigate('PostDetailPage', {
         feedId: notification.data.feed_id,
         is_from_pn: true
       });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The check for `notification?.data?.type === 'feed' || notification?.data?.type === 'reaction'` is good, but it would be better to also check if `notification.data.feed_id` exists before navigating to the 'PostDetailPage'. This will prevent potential crashes if `feed_id` is undefined.
```diff
-    if (notification?.data?.type === 'feed' || notification?.data?.type === 'reaction') {
+    if ((notification?.data?.type === 'feed' || notification?.data?.type === 'reaction') && notification?.data?.feed_id) {
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->